File: go/sum-of-multiples/sum_of_multiples.go
package summultiples

func SumMultiples(limit int, factors ...int) int {
	sum := 0
	for n := 1; n < limit; n++ {
		for _, f := range factors {
			if f != 0 && n % f == 0 {
				sum += n
				break
			}
		}
	}
	return sum
}

File: clojure/sum-of-multiples/src/sum_of_multiples.clj
(ns sum-of-multiples)

(defn isMultiple? [factors num]
  (some #(zero? (mod num %)) factors))

(defn sum-of-multiples [factors limit]
  (->> (range 1 limit)
       (filter #(isMultiple? factors %))
       (reduce +)
  ))
